ddd-koans/Value Object/Summary/src/Main.kt

```diff
@@ -0,0 +1,36 @@
+fun `which of the below concepts are connected with the Value Object?`(): Set<String> =
+    setOf(
+        "Immutability",
+        "Self-validation",
+        "Property-based equality",
+        "Domain-Driven Design"
+    )
+
+data class TimeSignature(private val numerator: Numerator, private val denominator: Denominator) {
+    companion object {
+        private val timeSignatureRegex = "^(\\d{1,2})/(\\d{1,2})$".toRegex()
+
+        fun create(numerator: Int, denominator: Int) =
+            TimeSignature(Numerator(numerator), Denominator(denominator))
+
+        fun of(timeSignature: String): TimeSignature {
+            val groupValues = timeSignatureRegex.matchEntire(timeSignature)?.groupValues
+            return groupValues?.let {
```

Comment:
personally, I find below snippet more readable that current 

```
timeSignatureRegex.matchEntire(timeSignature)
                ?.groupValues
                ?.let { group -> group[1] to group[2] }
                ?.let { (numerator, denominator) -> TimeSignature(numerator.toInt(), denominator.toInt()) }
                ?: throw RuntimeException("The numerator and denominator must be present")
```

Reply:
Alright, I will use the snippet and you are right about its readability 👍